server/raft.go

```diff
@@ -3383,35 +3383,37 @@ func (n *raft) processAppendEntry(ae *appendEntry, sub *subscription) {
 		n.updateLeadChange(false)
 	}
 
-	if (isNew && ae.pterm != n.pterm) || ae.pindex != n.pindex {
```

Comment:
Removing `isNew` makes sure we can also detect differences if we're catching up, otherwise we could wrongly assume our own state is correct.

Specifically covered by the test `TestNRGWALEntryWithoutQuorumMustTruncate/diverged`